src/components/form-component.js

```diff
@@ -98,12 +98,7 @@ function createFormClass(s = defaultStrategy) {
       if (this.props.getDispatch) {
         this.props.getDispatch(this.props.dispatch);
       }
-    }
-
-    UNSAFE_componentWillReceiveProps(nextProps) {
-      if (containsEvent(nextProps.validateOn, 'change')) {
-        this.validate(nextProps);
-      }
+      clearGetFormCacheForModel(this.props.model);
```

Comment:
This is the PR's main change -- by deferring this cache clearing to componentDidMount, we are no longer clearing cache on every change to the redux store (even ones that are not directly related to the form).